src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts

```diff
@@ -52,21 +57,36 @@ export class HaDeviceInfoZWaveJS extends LitElement {
     if (!this._nodeId || !this._entryId) {
       return;
     }
+
+    const configEntries = await getConfigEntries(this.hass);
+    this._configEntry = configEntries.find(
+      (entry) => entry.entry_id === this._entryId!
+    );
+
+    const zwaveJSIntegrations = configEntries.filter(
+      (entry) => entry.domain === "zwave_js"
+    );
```

Comment:
Let's filter first and then search in those results for the config entry :-)

Or do it all in 1 loop:

```ts
let zwaveJsConfEntries = 0;
for (const entry of configEntries) {
	if (entry.domain !== "zwave_js") {
		continue;
	}
	if (zwaveJsConfEntries) {
		this._multipleConfigEntries = true;
	}
	if (entry.entry_id === this._entryId!) {
		this._configEntry = entry;
	}
	if (this._configEntry && this._multipleConfigEntries) {
		break;
	}
}
```